### ISSUE TITLE
Test failure: 6.x branch testSqlFailsOnIndexWithTwoTypes

### ISSUE THREAD

**Issue by jakelandis** [elastic/elasticsearch#34774]
Reproduce on the 6.x branch:
```
./gradlew :x-pack:qa:full-cluster-restart:with-system-key:v5.6.13-SNAPSHOT#upgradedClusterTestRunner -Dtests.seed=6288B454803CE48E -Dtests.class=org.elasticsearch.xpack.restart.FullClusterRestartIT -Dtests.method="testSqlFailsOnIndexWithTwoTypes" -Dtests.security.manager=true -Dtests.locale=da -Dtests.timezone=Africa/Addis_Ababa -Dcompiler.java=11 -Druntime.java=8
```

```
FAILURE 0.86s | FullClusterRestartIT.testSqlFailsOnIndexWithTwoTypes <<< FAILURES!
   > Throwable #1: junit.framework.AssertionFailedError: Expected exception ResponseException but no exception was thrown
   >    at __randomizedtesting.SeedInfo.seed([6288B454803CE48E:C991DD8BECBEC63]:0)
   >    at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2693)
   >    at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2678)
   >    at org.elasticsearch.xpack.restart.FullClusterRestartIT.testSqlFailsOnIndexWithTwoTypes(FullClusterRestartIT.java:451)
   >    at java.lang.Thread.run(Thread.java:748)
```
git bisect points to https://github.com/elastic/elasticsearch/commit/6a99de4972e076d58cbc55ff33cd138c464e2531 as the culprit. cc @costin

**Comment by jtibshirani** [comment]
Muted the test in 9ab3bb7b24711ea60e34717bb7aed6e740e89ce7.

**Comment by costin** [comment]
I have investigated this issue which is caused by SQL not throwing an exception when dealing with indices with more than one type. This is cause by the move to the field caps API which gives a field view as oppose to a mapping view.
Since fields within types need to have the same mapping and we don't support types in the queries, this isn't really a problem.
I've raised an issue (see #34719) to reinstantiate the message and solve potential confusion of what happens to the underlying type.
